src/main/resources/script_templates/Neuroanatomy/Utilities/Filter_Multiple_Images.py

```diff
@@ -0,0 +1,97 @@
+# @boolean(label="Include subdirectories") recursive
+# @float(label="Size of structures to be filtered (spatially calibrated units)") scale
+# @LogService log
+# @StatusService status
+# @UIService uiservice
+
+"""
+file: Filter_Multiple_Images.py
+author: Tiago Ferreira, Cameron Arshadi
+info: Bulk filtering of image files using Frangi Vesselness
+"""
+
+import os
+
+
+def get_image_files(directory, filtering_string, extension):
+    """Returns a list containing the paths of files in the specified
+       directory. The list will only include files with the supplied
+       extension whose filename contains the specified string."""
+    files = []
+    for (dirpath, dirnames, filenames) in os.walk(directory):
+        for f in filenames:
+            if os.path.basename(f).startswith('.'):
+                continue
+            if filtering_string in f and f.lower().endswith(extension):
+                files.append(os.path.join(dirpath, f))
+        if not recursive:
+            break  # do not process subdirectories
+
+    return files
+
+
+def run():
+    # First check that scale parameter is > 0, exiting if not
+    if scale <= 0:
+        log.error('Please select a value > 0 for the scale parameter. Exiting...')
+        return
+
+    # Get all files with specified name filter and extension in the input directory
+    d = str(input_dir)
+    extension = ".tif"
+    files = get_image_files(d, name_filter, extension)
+    if not files or len(files) == 0:
+        uiservice.showDialog("No files matched the specified criteria", "Error")
+        return
+
+    processed = 0
+    skipped = 0
+    for f in files:
+
+        basename = os.path.basename(f)
+        msg = 'Processing file %s: %s...' % (processed + skipped + 1, basename)
+        status.showStatus(msg)
+        log.info(msg)
+
+        # Load the input image
+        input_image = ij.io().open(f)
+
+        # Verify that the image is 2D/3D and grayscale, skipping it if not
+        num_dimensions = input_image.numDimensions()
+        if num_dimensions > 3 or int(input_image.getChannels()) > 1:
+            log.error('Could not process %s...Only 2D/3D grayscale images are supported' % basename)
+            skipped += 1
+            continue
+
+        # Convert input image to float, since we are dealing with derivatives
+        float_input = ij.op().run("convert.float32", input_image)
+
+        # Obtain spatial calibration of the image
+        x_spacing = float_input.averageScale(0)
+        y_spacing = float_input.averageScale(1)
+        spacing = [x_spacing, y_spacing]
+        if num_dimensions == 3:
+            z_spacing = float_input.averageScale(2)
+            spacing.append(z_spacing)
+
+        # Create placeholder image for the output then run the Frangi Vesselness op
+        output = ij.op().run("create.img", float_input)
+        ij.op().run("frangiVesselness", output, float_input, spacing, scale)
```

Comment:
Yes, @tferr, this is correct.

Reply:
i see, so would the fix simply be to change ```# @float(label="Size of structures to be filtered (spatially calibrated units)") scale``` to ```# @float(label="Size of structures to be filtered (pixel units)") scale```? Or would the spacing need to be changed as well?

Reply:
`spacing` is used by the Op to refer to the physical distance between each sample, which you seem to calculate right. I also think that change to pixel units is sufficient.

Reply:
(Hopefully, you guys will not think I'm being anal, but I disagree).
it is _way_ more convenient to work on physical units. Take the following example for raster confocal microscopy: One acquires several images each containing a single cell (e.g., a neuron). Because cell size varies, one tweaks digital zoom so that a cell is compliteley included in field of view. The zoom adjustments respect the Nyquist criterion and are valid. One places all those images in a directory and tries to run this script. How would that work? Split the images by pixel size? "Eye-ball" a common value? I sincerely think, most users know the size of the features they want to enhance (e.g., these dendrites range between 6-10um, a soma  is ~10um, etc), but they may have no idea what to the _exact_  pixels that corresponds to. That info is stored in the image metadata. Furthermore, real-world coordinates are transferable across microscopy modalities and allow scripts to run as-is across labs. So, in short, this is what I would propose:
1) Keep the prompt in physical units
2) Compute the pixel size as the average of the voxel dimensions (one needs to take care that with 2D images, any existing metadata on depth (Z-dimension) is not averaged). Reducing voxel dimensions to this averaged isotropic dimension is a common, reasonable assumption.
3) Add a description to the prompt that this is what happening.

Reply:
I'm not sure I follow. Compute the ```scale``` parameter as the average of the voxel dimensions?

Reply:
or do that with the spacing and force it to be isotropic?

Reply:
The former:

```python
scale /= (sum(spacing) / len(spacing))  ## average voxel size: convert scale to pixel units
ij.op().run("frangiVesselness", output, float_input, spacing, scale)
```

Reply:
So then remove the input from the prompt and just say that's what's happening? Or allow them to input a multiplier for the later calculated "baseline"?

Reply:
oh i missed the /=, i see will fix it